augur/merge.py

```diff
@@ -407,6 +546,100 @@ def merge_metadata(
     db.cleanup()
 
 
+def get_sequences(input_sequences: List[str]):
+    try:
+        sequences = parse_inputs(input_sequences)
+    except InvalidNamedInputError as e:
+        raise AugurError(dedent(f"""\
+            Input filenames cannot start with '='.
+
+            The following {_n("input starts", "inputs start", len(e.invalid))} with '=':
+
+              {indented_list([repr(x) for x in e.invalid], '            ' + '  ')}
+            """)) from e
+    except DuplicateInputNameError as e:
+        raise AugurError(dedent(f"""\
+            Sequence input names must be unique.
+
+            The following {_n("name was", "names were", len(e.duplicates))} used more than once:
+
+              {indented_list([repr(x) for x in e.duplicates], '            ' + '  ')}
+            """)) from e
+
+    for name, path in sequences:
+        if name == "":
+            yield UnnamedSequenceFile(path)
+        else:
+            yield NamedSequenceFile(name, path)
```

Comment:
As an example of how to not let the internals of this function leak thru to its return type, consider:

```suggestion
    def sequence_files():
        for name, path in sequences:
            if name == "":
                yield UnnamedSequenceFile(path)
            else:
                yield NamedSequenceFile(name, path)

    return list(sequence_files())
```